src/snowflake/snowpark/modin/pandas/general.py

```diff
@@ -1585,7 +1585,12 @@ def to_datetime(
         )
     arg_is_scalar = is_scalar(arg)
     # handle empty array, list, dict
-    if not arg_is_scalar and not isinstance(arg, (DataFrame, Series)) and len(arg) == 0:
+    # TODO: SNOW-1372242: Remove instances of to_pandas when lazy index is implemented
+    if (
+        not arg_is_scalar
+        and not isinstance(arg, (DataFrame, Series))
+        and len(arg.to_pandas() if isinstance(arg, pd.Index) else arg) == 0
```

Comment:
If you haven't already prioritized this, let's add `Index.size` to the list of APIs we want to implement